### ISSUE TITLE
Release on AUR

### ISSUE THREAD

**Issue by TheLastGimbus** [localsend/localsend#40]
Hi there!

Found this app from "trending" - really like this so far, i think i will use it!

I would love to have it nicely installed with auto-updates so i can follow how you guys are going

Ekhm, I use Arch btw - and recently i've been uploading my own Dart app to AUR - the process is very simple: you create the account on AUR, make a PKGBUILD file with some fields, and then commit it to your app repo - to update, just make a new PKGBUILD and commit-push again :+1:

Here's the script that generates it for me:

https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/master/scripts/gen-pkgbuild.bash

It just downlaods the binary from gh releases

Here i use it in a gh action:

https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/3a0d48e8555ab0c20c8da12424d43bb7888f4516/.github/workflows/new-release.yml#L72-L75

Of course, it's simpler for me because it's a single binary, but you can also look at how @KRTirtho does it in https://github.com/KRTirtho/spotube - they have a really great setup with all the releases :+1:

**Comment by Tienisto** [comment]
Hi thanks! I will take a look.

**Comment by msarkisian** [comment]
I'd appreciate this a ton as well.
Thanks for the application!

**Comment by RotationMatrix** [comment]
I wrote this PKGBUILD for myself and it seems to work so far. Might be useful as a start. [`flutter`](https://aur.archlinux.org/packages/flutter) needs to be installed from the AUR prior to building for it to work.

```
pkgname=localsend
pkgver=1.6.1
pkgrel=1
pkgdesc="An open source cross-platform alternative to AirDrop"
arch=('x86_64')
url="https://github.com/localsend/localsend"
license=('MIT')
groups=()
depends=()
makedepends=('flutter' 'dart')
optdepends=()
provides=('localsend')
conflicts=()
replaces=()
backup=()
options=()
install=
changelog=
source=("https://github.com/localsend/localsend/archive/refs/tags/v$pkgver.tar.gz"
        "https://raw.githubusercontent.com/localsend/localsend/v$pkgver/LICENSE"
        "localsend.desktop")
noextract=()
sha256sums=('4515f8d6c9051f243cade3d7c6c659644b328aabae69530dc87634f56e1517c5'
            'c483a1de828cbeebce8c21aa0d66f23a5783cc381793f2ca5d172a9cd4ed8f4d'
            '82c33b4c5e56e78beb22c10ba122f51af51fee52c5ae1b665fd8e10053287e8f')

build() {
  cd "$pkgname-$pkgver"

  sed -i "s/collection: 1.16.0/collection: 1.17.0/" pubspec.yaml

  flutter pub get
  flutter pub run build_runner build
  flutter build linux
}

package() {
  cd "$pkgname-$pkgver"

  mkdir -p "$pkgdir/opt"
  cp -r build/linux/x64/release/bundle "$pkgdir/opt/$pkgname"

  install -Dm644 assets/img/logo-512.png "$pkgdir/usr/share/pixmaps/localsend.png"
  install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
  install -Dm755 "$srcdir/localsend.desktop" "$pkgdir/usr/share/applications/localsend.desktop"
}
```

And the contents of `localsend.desktop`:
```
#!/usr/bin/env xdg-open
[Desktop Entry]
Categories=Internet;
Comment=An open source cross-platform alternative to AirDrop
Exec=/opt/localsend/localsend_app
Icon=localsend.png
MimeType=
Name=LocalSend
StartupWMClass=localsend_app
Terminal=false
Type=Application
```

**Comment by TheLastGimbus** [comment]
Some notes if you're not familiar with arch/aur stuff @Tienisto 

arch has a default repository, where all packages are (tl;dr) downloadable binaries, as they should be - but only long-lived trusted software gets there

so there's the aur, which doesn't contain any binaries themselves, but just PKGBUILD files, which are instructions for `pacman` on how to build a package

so, a package named `localsend` should contain pkgbuild that has `flutter` in `makedepends` (dependencies required to build, and then builds `localsend` itself. This is nice, but could potentially take hours for some users

so, there is a trend that you name a package `yourpackage-bin`, and pkgbuild for those just downloads the binaries from server

# SO

the best for `localsend` would probably to also include `linux.tar.gz` versions of the app in the gh releases, and then tell pkgbuild to download and export it

# :+1:

**Comment by Tienisto** [comment]
Thanks for your inputs @TheLastGimbus , @RotationMatrix .

That's good to know. The solution by @RotationMatrix is good but it uses the latest flutter version instead of the version provided by fvm ([3.3.10](https://github.com/localsend/localsend/blob/main/.fvm/fvm_config.json)). In most cases, it works though.

Will it just work if I publish the result from `flutter build linux`? I think Flutter depends on gtk-something but I am not sure.

Some users have issues with the AppImage currently: #14, #30.

**Comment by TheLastGimbus** [comment]
> Will it just work if I publish the result from flutter build linux

Yeah, that's how `localsend-bin` would work :+1:

Just upload `.tar.gz` to releases, name them nicely with `v{semantic.version.ing}`, and use it as source

Then few lines in `package()` to install .desktop files and you're good

again, look up how spotube does it: https://github.com/KRTirtho/spotube/blob/master/aur-struct/PKGBUILD

He got it a bit more complicated - if you want, to you can do it simillary to me - for example, i just get the version with `grep`: https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/3a0d48e8555ab0c20c8da12424d43bb7888f4516/scripts/gen-pkgbuild.bash?rgh-link-date=2023-01-28T12%3A09%3A14Z#L5

**Comment by RotationMatrix** [comment]
> it uses the latest flutter version instead of the version provided by fvm ([3.3.10](https://github.com/localsend/localsend/blob/main/.fvm/fvm_config.json)).

Ah, maybe that is why the build needed `collection: 1.17.0` to work? I had a feeling the AUR flutter was newer than that used for LocalSend, but wasn't sure.

> Will it just work if I publish the result from `flutter build linux`?

That would be perfect! Then we can eliminate the flutter build dependency altogether.

**Comment by Tienisto** [comment]
Well, someone already published a package. I don't know if a real publisher is needed

**Comment by TheLastGimbus** [comment]
Oh... interesting

Publishing straight from gh actions would definitely be faster, but maybe this guy is good enough 👍

We can always email him if we wanted to take over 💌

**Comment by Nixuge** [comment]
Discovered through trending too and figured out i'd made a quick AUR package (ended up making a -bin and a -git), if an actual maintainer of the project wants to take over or just make a better pkgbuild feel free to

**Comment by Nixuge** [comment]
Aight well first update I have to do and first update where I'm late lol, outdated since this morning and coming back home only tonight, sorry for that

**Comment by sikmir** [comment]
`localsend` is available on NixOS as well now.

**Comment by Tienisto** [comment]
@Nixuge

Since `v1.10.0`, LocalSend also provides `.tar.gz`. Using this is more lightweight and it should provide faster startup times.

I recently migrated the flatpak to use `.tar.gz`: https://github.com/flathub/org.localsend.localsend_app

Maybe it is possible on Arch too? I don't know if you could fix the `libayatana-appindicator` though.

**Comment by Nixuge** [comment]
Hello @Tienisto, see https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=localsend-bin#n9
(Was about the deb but same problem w the targz)

Nothing I can do until either LocalSend or the lib on arch's repos gets fixed

**Comment by Nixuge** [comment]
> nothing I can do

More like nothing I can do except redistributing Debian's lib, which is something I'd really like to avoid

If things remain like that for too much time will consider doing so anyways

**Comment by TheLastGimbus** [comment]
Hey umm, about this - i wanted to install this recently, and...

![image](https://github.com/localsend/localsend/assets/40139196/e78ff1bc-b300-4909-89d1-7373aaeb351f)

there are four packages, from 3 different people, 2 of which are outdated. Mayyybe we could unifiy at least the `-bin` package to be updated by github action on this repo?

Again, I invite to steal from my action cause I think I got it nice :ok_hand: :
- https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/0e28e5bb2d1bcc9a4bbdbbbaed35648533c088c4/.github/workflows/new-release.yml#L72-L86 
- https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/master/scripts/gen-pkgbuild.bash

**Comment by Nixuge** [comment]
Hello,

The -git package is up to date as -git packages update automatically

The -bin package however has been quite a pain, it's always been an AppImage, but I've been wanting to use the provided .deb/.tar.gz for quite a long time...
...except as of 1.9.0 there's a hard issue with libayatana-appindicator on arch that caused the program to segfault instantly if not using the provided libs (see #409)
Honestly not sure what to do with it, could just have it use the deb/targz and [download the dependency from another source](https://packages.debian.org/sid/amd64/libayatana-appindicator3-1/download), but that'd honestly be quite suboptimal, altho possible
If anyone wants to/manages to tackle that issue without downloading from third party websites, feel free to dm me anywhere and i'll give you owner/coowner whathever you want

tldr; the -bin is just an appimage with all libs included because Arch's libayatana-appindicator instantly segfaults otherwise
